airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/format/check/python/commands.py

```diff
@@ -0,0 +1,69 @@
+import logging
+import sys
+from typing import Optional
+
+import asyncclick as click
+import dagger
+from pipelines.airbyte_ci.format.consts import DEFAULT_FORMAT_IGNORE_LIST
+from pipelines.cli.click_decorators import LazyPassDecorator, click_ignore_unused_kwargs
+from pipelines.helpers.utils import sh_dash_c
+from pipelines.models.contexts.click_pipeline_context import ClickPipelineContext
+
+pass_pipeline_context: LazyPassDecorator = LazyPassDecorator(ClickPipelineContext)
+
+
+@click.command()
+@pass_pipeline_context
+@click_ignore_unused_kwargs
+async def python(ctx: ClickPipelineContext, dagger_client: Optional[dagger.Client] = None):
+    """Format python code via black and isort."""
+    success = await format_python(ctx, dagger_client)
+    if not success:
+        click.Abort()
+
+
+async def format_python(ctx: ClickPipelineContext, dagger_client: Optional[dagger.Client] = None) -> bool:
+    """Checks whether the repository is formatted correctly.
+    Args:
+        fix (bool): Whether to automatically fix any formatting issues detected.
+    Returns:
+        bool: True if the check/format succeeded, false otherwise
+    """
+    logger = logging.getLogger(f"format")
+
+    dagger_client = ctx.params["dagger_client"]
+    try:
+        format_container = await (
+            dagger_client.container()
+            .from_("python:3.10.13-slim")
```

Comment:
Seconding this and it would be nice to be able to reuse many of the same top-level layers so that we don't have to `apt-get update` and friends  more than we have to. Not a blocking comment, just a nice to-have.

Reply:
> local and remote execution of the format command

I'm not sure how this changes, local and remote formatting should always use the same image either way since they are run with the same command. (Since we no longer format in connectors test). Can you elaborate? 

Happy to switch to whatever python version, we sure have a lot of them across the codebase though.

Reply:
> so that we don't have to apt-get update and friends more than we have to

@postamar You're right, we definitely don't have to do that! removed that already.